tests/unittests/CoreText/CTFontTests.mm

```diff
@@ -513,4 +515,203 @@ virtual void TearDown() {
+                EXPECT_NEAR(comparePathContext->expectedElements[comparePathContext->count].point.x, element->points[2].x, 0.002f);
+                EXPECT_NEAR(comparePathContext->expectedElements[comparePathContext->count].point.y, element->points[2].y, 0.002f);
+                break;
+
+            case kCGPathElementCloseSubpath:
+                // No-op, CloseSubpath has no associated point
+                break;
+
+            default:
+                EXPECT_TRUE_MSG(false, @"An invalid CGPathElementType was returned from CTFontCreatePathForGlyph");
+                break;
+        }
+        ++comparePathContext->count;
+    };
+
+    // Iterate through all elements in the path, check against expected elements
+    CGPathApply(unscaledPath, &comparePathContext, comparePathToExpected);
+
+    // Make sure that all of expectedElements has been iterated over
+    ASSERT_EQ(expectedElements.size(), comparePathContext.count);
+
+    // Introduce font size
+    CGFloat fontSize = 13;
+    const CGAffineTransform scaleByFontSize = CGAffineTransformMakeScale(fontSize / unitsPerEm, fontSize / unitsPerEm);
+
+    font = CTFontCreateWithName(c_timesNewRomanName, fontSize, nullptr);
+    CFAutorelease(font);
+
+    CGPathRef pathWithFontSize = CTFontCreatePathForGlyph(font, capitalRGlyph, nullptr);
+    CFAutorelease(pathWithFontSize);
+
+    // Compare against a transformed version of expectedElements
+    comparePathContext = { 0, std::vector<ExpectedCGPathElement>(expectedElements.size()) };
+    std::transform(expectedElements.begin(),
+                   expectedElements.end(),
+                   comparePathContext.expectedElements.begin(),
+                   [scaleByFontSize](ExpectedCGPathElement element) {
+                       return ExpectedCGPathElement{ element.type, CGPointApplyAffineTransform(element.point, scaleByFontSize) };
+                   });
+    CGPathApply(pathWithFontSize, &comparePathContext, comparePathToExpected);
+    ASSERT_EQ(expectedElements.size(), comparePathContext.count);
+
+    // Introduce a transform on the font
+    const CGAffineTransform fontTransform =
+        CGAffineTransformTranslate(CGAffineTransformScale(CGAffineTransformMakeRotation(0.5), 6, -3), 10, 20); // Arbitrary transform
+
+    font = CTFontCreateWithName(c_timesNewRomanName, fontSize, &fontTransform);
+    CFAutorelease(font);
+
+    CGPathRef pathWithFontSizeAndFontTransform = CTFontCreatePathForGlyph(font, capitalRGlyph, nullptr);
+    CFAutorelease(pathWithFontSizeAndFontTransform);
+
+    // Concat the font size matrix and the non-translation portions of the fontTransform to get the expected path
+    CGAffineTransform fontTransformWithoutTranslate = fontTransform;
+    fontTransformWithoutTranslate.tx = 0;
+    fontTransformWithoutTranslate.ty = 0;
+    CGAffineTransform totalTransform = CGAffineTransformConcat(fontTransformWithoutTranslate, scaleByFontSize);
+
+    // Compare against a transformed version of expectedElements
```

Comment:
>a transformed version of expectedElements [](start = 23, length = 41)

i wanted to use CGPathCreateCopyByTransformingPath, but we don't have that yet :(